pkg/controller/elasticsearch/driver/upscale.go

```diff
@@ -43,7 +40,12 @@ func HandleUpscaleAndSpecChanges(
 	expectedResources nodespec.ResourcesList,
 ) (sset.StatefulSetList, error) {
 	// adjust expected replicas to control nodes creation and deletion
-	adjusted, err := adjustResources(ctx, actualStatefulSets, expectedResources)
+	upscaleState, err := newUpscaleState(ctx, actualStatefulSets, expectedResources)
```

Comment:
Ah, ok. From one of our conversations I understood that initonce is redundant as we do it every time, so it can go away while behavior can stay. Will fix.